src/components/MultiHopTrade/components/TradeConfirm/hooks/useMixpanel.tsx

```diff
@@ -4,15 +4,20 @@ import { getMixpanelEventData } from '@/components/MultiHopTrade/helpers'
 import { getMixPanel } from '@/lib/mixpanel/mixPanelSingleton'
 import type { MixPanelEvent } from '@/lib/mixpanel/types'
 
-export const useMixpanel = () => {
+export const useMixpanel = (allowUndefinedEventData = false) => {
   const mixpanel = useMemo(() => getMixPanel(), [])
-  const eventData = useMemo(() => getMixpanelEventData(), [])
   const trackMixpanelEvent = useCallback(
     (event: MixPanelEvent) => {
       // mixpanel is undefined when the feature is disabled
-      if (eventData && mixpanel) mixpanel.track(event, eventData)
+      if (!mixpanel) return
+
+      const eventData = getMixpanelEventData()
+      // Allow tracking without eventData if flag is set
```

Comment:
q: is there any benefit in supporting an optional `allowUndefinedEventData`?

Feels like since this is a wrapper over `mixpanel.track()`, we could simply remove the `eventData` to comply with the signature of the `track()` method, WDYT?

<img width="1576" height="71" alt="image" src="https://github.com/user-attachments/assets/78ac7819-72a9-4f08-b4ab-c36d863ee95d" />